modules/store/src/provide_store.ts

```diff
@@ -0,0 +1,333 @@
+    | ActionReducer<any, any>
+    | InjectionToken<ActionReducer<any, any>>,
+  config: StoreConfig<any, any> | InjectionToken<StoreConfig<any, any>> = {}
+): ImportedNgModuleProviders {
+  return {
+    ɵproviders: [
+      ..._provideState(featureNameOrSlice, reducers, config),
+      ENVIRONMENT_STATE_PROVIDER,
+    ],
+  };
+}
+
+export function _provideStore(
+  reducers:
+    | ActionReducerMap<any, any>
+    | InjectionToken<ActionReducerMap<any, any>>,
+  config: RootStoreConfig<any, any>
+) {
+  return [
+    {
+      provide: _ROOT_STORE_GUARD,
+      useFactory: _provideForRootGuard,
+      deps: [[Store, new Optional(), new SkipSelf()]],
+    },
+    { provide: _INITIAL_STATE, useValue: config.initialState },
+    {
+      provide: INITIAL_STATE,
+      useFactory: _initialStateFactory,
+      deps: [_INITIAL_STATE],
+    },
+    { provide: _INITIAL_REDUCERS, useValue: reducers },
+    {
+      provide: _STORE_REDUCERS,
+      useExisting:
+        reducers instanceof InjectionToken ? reducers : _INITIAL_REDUCERS,
+    },
+    {
+      provide: INITIAL_REDUCERS,
+      deps: [Injector, _INITIAL_REDUCERS, [new Inject(_STORE_REDUCERS)]],
+      useFactory: _createStoreReducers,
+    },
+    {
+      provide: USER_PROVIDED_META_REDUCERS,
+      useValue: config.metaReducers ? config.metaReducers : [],
+    },
+    {
+      provide: _RESOLVED_META_REDUCERS,
+      deps: [META_REDUCERS, USER_PROVIDED_META_REDUCERS],
+      useFactory: _concatMetaReducers,
+    },
+    {
+      provide: _REDUCER_FACTORY,
+      useValue: config.reducerFactory ? config.reducerFactory : combineReducers,
+    },
+    {
+      provide: REDUCER_FACTORY,
+      deps: [_REDUCER_FACTORY, _RESOLVED_META_REDUCERS],
+      useFactory: createReducerFactory,
+    },
+    ACTIONS_SUBJECT_PROVIDERS,
+    REDUCER_MANAGER_PROVIDERS,
+    SCANNED_ACTIONS_SUBJECT_PROVIDERS,
+    STATE_PROVIDERS,
+    STORE_PROVIDERS,
+    provideRuntimeChecks(config.runtimeChecks),
+    checkForActionTypeUniqueness(),
+  ];
+}
+
+/**
+ * Environment Initializer used in the root
+ * providers to initialize the Store
+ */
+const ENVIRONMENT_STORE_PROVIDER = [
+  {
+    provide: ENVIRONMENT_INITIALIZER,
+    multi: true,
+    useFactory() {
+      return () => {
+        inject(ActionsSubject);
+        inject(ReducerObservable);
+        inject(ScannedActionsSubject);
+        inject(Store);
+        inject(_ROOT_STORE_GUARD, InjectFlags.Optional);
+        inject(_ACTION_TYPE_UNIQUENESS_CHECK, InjectFlags.Optional);
+      };
+    },
+  },
+];
+
+/**
+ * Provides the global Store providers and initializes
+ * the Store.
+ * These providers cannot be used at the component level.
+ *
+ * @usageNotes
+ *
+ * ### Providing the Global Store
+ *
+ * ```ts
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideStore({})],
+ * });
+ * ```
+ */
+export function provideStore<T, V extends Action = Action>(
+  reducers: ActionReducerMap<T, V> | InjectionToken<ActionReducerMap<T, V>>,
```

Comment:
Could we make reducers optional?
Reason: I like to just provide the store, and use features/state to build up the state tree. Otherwise, I'm worried that state end up in `provideStore` that doesn't belong there. And `provideStore()` looks better than `provideStore({})` imho.

```suggestion
  reducers?: ActionReducerMap<T, V> | InjectionToken<ActionReducerMap<T, V>>,
```

Reply:
Good catch. We've been trying to get to this point by default for a while where nothing is registered in the root